cosmoz-image-viewer.html

```diff
@@ -594,123 +549,153 @@ <h2>An error occurred while loading the image.</h2>
 			</div>
 			<script>
 				/*eslint-env browser*/
-				const ciw = {
-					actions: {
+				(() => {
+					class CIV {
+						constructor(images = []) {
+							this._images = images;
+							this._currentImageIndex = 0;
+							this.elements = [
+								'image',
+								'loading',
+								'prevImg',
+								'printContainer',
+								'nextImg',
+								'pageNumber',
+								'actions',
+								'navs',
+								'error'
+							].reduce((elements, id) => Object.assign(elements, { [id]: document.getElementById(id) }), {});
+							if (this.elements.image == null) {
+								return;
+							}
+							this.elements.image.onload = () => this._setDisplay('loading', false);
+							this.elements.image.onerror = () => this._setDisplay('error', true);
+						}
+
+						get isLoading() {
+							return this._isLoading;
+						}
+
+						set isLoading(loading) {
+							this._isLoading = true;
+							this._setDisplay('loading', loading);
+						}
+
+						set currentImageHasError(hasError) {
+							this._currentImageHasError = hasError;
+						}
+
+						get currentImageIndex() {
+							return this._currentImageIndex;
+						}
+
+						set currentImageIndex(index) {
+							this._currentImageIndex = index;
+							this._changeCurrentImage(this.images[index]);
+							this._setDisplay('error', false);
+							this._updateNavState();
+						}
+
+						set images(images) {
+							this._images = images;
+							this.currentImageIndex = 0;
+							this._setDisplay('actions', images.length > 0, 'flex');
+							this._setNavVisibility(images.length > 1);
+						}
+
+						get images() {
+							return this._images;
+						}
+
+						setImages(images, currentImageIndex) {
```

Comment:
Actually, if we need our own setter anyway, we should just use this one with a default index of 0.